enterprise/server/hit_tracker_client/hit_tracker_client.go

```diff
@@ -2,57 +2,307 @@ package hit_tracker_client
 
 import (
 	"context"
+	"flag"
+	"sync"
+	"sync/atomic"
 	"time"
 
-	"github.com/buildbuddy-io/buildbuddy/server/environment"
 	"github.com/buildbuddy-io/buildbuddy/server/interfaces"
+	"github.com/buildbuddy-io/buildbuddy/server/metrics"
 	"github.com/buildbuddy-io/buildbuddy/server/real_environment"
+	"github.com/buildbuddy-io/buildbuddy/server/util/alert"
+	"github.com/buildbuddy-io/buildbuddy/server/util/authutil"
+	"github.com/buildbuddy-io/buildbuddy/server/util/claims"
+	"github.com/buildbuddy-io/buildbuddy/server/util/grpc_client"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/buildbuddy-io/buildbuddy/server/util/proto"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+	"github.com/prometheus/client_golang/prometheus"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/durationpb"
 
+	hitpb "github.com/buildbuddy-io/buildbuddy/proto/hit_tracker"
 	repb "github.com/buildbuddy-io/buildbuddy/proto/remote_execution"
+	rspb "github.com/buildbuddy-io/buildbuddy/proto/resource"
+	gstatus "google.golang.org/grpc/status"
 )
 
+const (
+	// The initial capacity to allocate for slices storing per-group cache
+	// hits. This is intended to be large enough to avoid too many allocations
+	// due to  slice growth, but not so large to incur excessive memory use.
+	initialSliceCapacity = 4_096
```

Comment:
Did this come from a profile? (If not, might want to remove this until we know that allocations are an issue - this might do more harm than good if we're e.g. using <50% of the buffer capacity most of the time)

Reply:
Nope, got rid of it.